patroni/api.py

```diff
@@ -89,8 +90,17 @@ def do_GET(self, write_status_code_only=False):
         patroni = self.server.patroni
         cluster = patroni.dcs.cluster
 
-        replica_status_code = 200 if not patroni.noloadbalance and \
-            response.get('role') == 'replica' and response.get('state') == 'running' else 503
+        replica_status_code = 503
+        if response.get('role') == 'replica' and response.get('state') == 'running' \
+           and not patroni.noloadbalance:
+            max_replica_lag = self.path_query.get('lag') and parse_int(self.path_query.get('lag')[0]) or 0
+            leader_optime = cluster and cluster.last_leader_operation and parse_int(cluster.last_leader_operation) or 0
+            if max_replica_lag and leader_optime:
```

Comment:
Since the `leader_optime` is cached, the "calculated" lag naturally quite often would be negative.
In theory someone might want to use `?lag=-100` in the health-check. I agree, that having the negative number makes little sense, but having `?lag=0` would be absolutely valid case.

Also I have a suggestion that we should always take lag into account and in case if it is not specified (or not valid) in the health-check assume that `lag = sys.maxsize`.

Reply:
True. We should allow lag=0 or -ve lag health checks. Will update the code.

Not sure if we should make the lag health-check default. Agree, we could use default/max-lag if incorrect value is specified? Please let me know your thoughts. Thanks.